http-support/src/main/java/org/eclipse/sw360/antenna/http/impl/HttpClientImpl.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) Bosch.IO GmbH 2020.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.sw360.antenna.http.impl;
```

Comment:
Actually,I was not sure whether there will only be a single implementation. A design goal of the HTTP library was to provide an abstraction from a concrete client library. When used outside of the Antenna context it may well be the case that a different HTTP client library is in use, e.g. Apache Http Client or the Java 9+ HTTP client. Then it is at least in theory possible that the interfaces are implemented on top of this other library (as they are not too complex).

However, supporting this scenario would require more work to separate the implementation into another module.

Reply:
I do not think that a meta framework is really needed here, right. I think a small use case specific framework that allows to make easy use of a REST API with HATEOAS implemented in the technology we think appropriate would do it for now. Btw, you can still implement this based on a different technology, you are only carrying the dead code in the package, which you do in the current implementation as well, since as you mentioned, the implementation is not in another module.

My principle concern is, that you can get an awful lot of advantages with restricting access to classes if they are packaged right. The common style of making many packages contradicts this advantage and this makes me feel sad.

Reply:
The api and impl packages have been removed. The classes are now in the top-level http package.